src/vehicle_autodrive.cpp

```diff
@@ -0,0 +1,960 @@
+    half_open_rectangle<point> ret;
+    if( omt_diff.x() < 0 ) {
+        ret.p_min.x = 0;
+        ret.p_max.x = NAV_MAP_PADDING;
+    } else if( omt_diff.x() == 0 ) {
+        ret.p_min.x = NAV_MAP_PADDING;
+        ret.p_max.x = OMT_SIZE + NAV_MAP_PADDING;
+    } else {
+        ret.p_min.x = OMT_SIZE + NAV_MAP_PADDING;
+        ret.p_max.x = OMT_SIZE + 2 * NAV_MAP_PADDING;
+    }
+    if( omt_diff.y() < 0 ) {
+        ret.p_min.y = 0;
+        ret.p_max.y = NAV_MAP_PADDING;
+    } else if( omt_diff.y() == 0 ) {
+        ret.p_min.y = NAV_MAP_PADDING;
+        ret.p_max.y = OMT_SIZE + NAV_MAP_PADDING;
+    } else {
+        ret.p_min.y = OMT_SIZE + NAV_MAP_PADDING;
+        ret.p_max.y = OMT_SIZE + 2 * NAV_MAP_PADDING;
+    }
+    return ret;
+}
+
+static half_open_rectangle<point> compute_nav_zone( const half_open_rectangle<point> &goal_zone )
+{
+    half_open_rectangle<point> ret;
+    ret.p_min.x = std::min( goal_zone.p_min.x, NAV_MAP_PADDING );
+    ret.p_min.y = std::min( goal_zone.p_min.y, NAV_MAP_PADDING );
+    ret.p_max.x = std::max( goal_zone.p_max.x, OMT_SIZE + NAV_MAP_PADDING );
+    ret.p_max.y = std::max( goal_zone.p_max.y, OMT_SIZE + NAV_MAP_PADDING );
+    return ret;
+}
+
+static int signum( int val )
+{
+    return ( 0 < val ) - ( val < 0 );
+}
+
+vehicle_profile vehicle::autodrive_controller::compute_profile( orientation facing ) const
+{
+    vehicle_profile ret;
+    tileray tdir( units::from_degrees( facing * TURNING_INCREMENT ) );
+    ret.tdir = tdir;
+    std::map<int, std::pair<int, int>> extent_map;
+    const point pivot = driven_veh.pivot_point();
+    for( const vehicle_part &part : driven_veh.parts ) {
+        if( part.removed ) {
+            continue;
+        }
+        tripoint pos;
+        vehicle::coord_translate( tdir, pivot, part.mount, pos );
+        if( extent_map.find( pos.y ) == extent_map.end() ) {
+            extent_map[pos.y] = { pos.x, pos.x };
+        } else {
+            auto &extent = extent_map[pos.y];
+            extent.first = std::min( extent.first, pos.x );
+            extent.second = std::max( extent.second, pos.x );
+        }
+    }
+    for( const auto &extent : extent_map ) {
+        const int y = extent.first;
+        const int x_min = extent.second.first;
+        const int x_max = extent.second.second;
+        for( int x = x_min; x <= x_max; x++ ) {
+            ret.occupied_zone.emplace_back( x, y );
+        }
+    }
+    for( int part_num : driven_veh.rotors ) {
+        const vehicle_part &part = driven_veh.part( part_num );
+        const int diameter = part.info().rotor_diameter();
+        const int radius = ( diameter + 1 ) / 2;
+        if( radius > 0 ) {
+            tripoint pos;
+            vehicle::coord_translate( tdir, pivot, part.mount, pos );
+            for( tripoint pt : points_in_radius( pos, radius ) ) {
+                ret.occupied_zone.emplace_back( pt.xy() );
+            }
+        }
+    }
+    tdir.advance( 10 );
```

Comment:
Please replace this magic number with a documented constexpr

Reply:
Not sure what to call it. It's just an arbitrary number that makes the tileray advance at least one step along each axis unless it's axis-aligned.
I could derive the "offset" I need in other ways, but it would be significantly more code than these 2 lines.

Reply:
"minimal_tileray_setp" and a comment about it would be very helpful, then.

Actually, an even better plan would be to set a default step of 10 and document why that is the default step in the tileray header.

Reply:
Hmm, I probably did not explain very well. The default step for tileray should stay at 1. I'm only incrementing by 10 here because when driving at a 15 degree angle most of the time you're only getting an increment along one axis and I want to know if there would eventually be an increment in the other axis as well (and whether these increments are positive or negative).
I added more comments so have a look to see if it makes more sense now.